### PR TITLE
Refactor smoreo

### PR DESCRIPTION
**Pull request by mo-sameh**

### CODE REVIEW

**Review by RoniEmad — CHANGES_REQUESTED** [review]

File: perception/smoreo/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+>Please notice that the point $(u,v,f)$ is in the camera coordinate system and not the image coordinate system, so we have to transform pixels to the camera frame first.
+## Dependencies
+* [ROS](http://wiki.ros.org/kinetic/Installation/Ubuntu)
+* [OpenCV](https://docs.opencv.org/3.4.3/d7/d9f/tutorial_linux_install.html)
+* [Numpy](https://pypi.org/project/numpy/)
+* [Matplotlib](https://pypi.org/project/matplotlib/)
+## Installation
+### Install ROS
+Follow the instructions on the [ROS Wiki](http://wiki.ros.org/kinetic/Installation/Ubuntu) to install ROS Kinetic.
```
Comment (ahmedemad242) [inline perception/smoreo/README.md:27]: We are using ROS noetic

File: perception/smoreo/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+*Please note that currently the tuned paramters are not saved, so manually change the parameters in the launch file after tuning the pipeline
+### Subscribe to the output topic
+The pipeline publishes the results to the `/perception/smoreo/detected_cones_lm`    topic. Subscribe to this topic to get the results.
+*To visualize the landmark array on rviz, use can use our MarkerViz package that parse landmarks to makers that can be easily visualized in rviz.*
+
+
+
+## License
```
Comment (ahmedemad242) [inline perception/smoreo/README.md:74]: License is included in the root directory no need to include it here

File: perception/smoreo/launch/smoreo_sys_flir.launch [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+
+
+
+	<!-- Nodes to be launched -->
+
+
+
+	<!--<include file="$( find bounding_box_filter)/launch/bounding_box_filter.launch"/>-->
```
Comment (ahmedemad242) [inline perception/smoreo/launch/smoreo_sys_flir.launch:15]: remove unnecessary comments

File: perception/smoreo/launch/smoreo_sys_ipg.launch [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<launch>
+
+	<node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera" args="0 0 0.4 0 0 0 base_link camera_link"/>
+	<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_image_coordinate_system" args="0 0 0 0.5 -0.5 0.5 0.5 worldcoordinates cameracoordinates"/>
+	<!-- Nodes to be launched -->
+	<!--<include file="$( find bounding_box_filter)/launch/bounding_box_filter.launch"/>-->
```
Comment (ahmedemad242) [inline perception/smoreo/launch/smoreo_sys_ipg.launch:7]: remove unnecessary comments

File: perception/smoreo/src/__init__.py
Comment (ahmedemad242) [inline perception/smoreo/src/__init__.py:1]: There should be no init.py in the src folder

File: perception/smoreo/package.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  <exec_depend>roscpp</exec_depend>
+  <exec_depend>dynamic_reconfigure</exec_depend>
+  <exec_depend>rospy</exec_depend>
+  <exec_depend>std_msgs</exec_depend>
+  <exec_depend>message_runtime</exec_depend>
+  <exec_depend>tf_helper</exec_depend>
+
+
+  <!-- The export tag contains other, unspecified, tags -->
```
Comment (ahmedemad242) [inline perception/smoreo/package.xml:34]: remove unnecessary comments

File: perception/smoreo/testing/generateTestCase.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+            [[720, 0.0, 720], [0.0, 720, 560], [0.0, 0.0, 1.0]],
+            dtype=np.float64,
+        ),
+        "worldCords_inCamera": np.array([[0, -1, 0], [0, 0, -1], [1, 0, 0]], dtype=int),
+        "camera_height_from_ground": 0.816,
+        "cut_off_y": 900.0,
+        "cone_height": 0.38,
+    }
+
+    BAG_PATH = "/media/mosameh/CEFE-EBFE/first_Ipg_moreo_inputs.bag"
```
Comment (ahmedemad242) [inline perception/smoreo/testing/generateTestCase.py:189]: you cannot use absolute paths from your system, please use relative paths

File: perception/smoreo/tests/__init__.py
Comment (ahmedemad242) [inline perception/smoreo/tests/__init__.py:1]: this is not a module there is no need for an __init__.py file

File: perception/smoreo/tests/smoreo_test.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        box2 = np.array([7.0, 8.0, 9.0, 10.0, 11.0, 12.0], dtype=np.float32)
+        smoreo.addToLandmarkArray(pose2, box2)
+        self.assertEqual(len(smoreo.allLandMarks.landmarks), 2)
+
+    def testPredictWithConeBase(self) -> None:
+        """
+        Test if the prediction is correct with cone base
+        """
+        with open(
+            r"/home/mosameh/fs-system2/src/perception/smoreo/testing/testCase1.pickle",
```
Comment (ahmedemad242) [inline perception/smoreo/tests/smoreo_test.py:173]: you cannot use absolute paths from your system, please use relative paths

**Review by ahmedemad242 — CHANGES_REQUESTED** [review]

**Review by RoniEmad — DISMISSED** [review]
Good for me

**Review by RoniEmad — DISMISSED** [review]
LGTM

**Review by ahmedemad242 — DISMISSED** [review]
LGTM! Good Job @mo-sameh @ranaamr22 

I am waiting for confirmation to merge

**Review by ahmedemad242 — APPROVED** [review]

**Pull request merged**